docs/docs/intro/configure.md

```diff
@@ -375,24 +375,121 @@ marathon {
-example.
+### Execution strategy
+When executing tests with retries there are multiple trade-offs to be made. Two execution strategies are supported: any success or all success.
+By default, any success strategy is used with fast execution i.e. if one of the test retries succeeds then the test is considered successfully
+executed and all non-started retries are removed.
+
+#### Any success
+Test passes if any of executions is passing. This mode works only if there is no complex sharding strategy applied. This is the default.
+
+<Tabs>
+<TabItem value="YAML" label="Marathonfile">
+
+```yaml
+executionStrategy:
+  mode: ANY_SUCCESS
+```
+
+</TabItem>
+<TabItem value="kts" label="Kotlin DSL">
+
+```kotlin
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ANY_SUCCESS)
+}
+```
+
+</TabItem>
+<TabItem value="groovy" label="Groovy DSL">
+
+```groovy
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ANY_SUCCESS)
+}
+```
+
+</TabItem>
+</Tabs>
+
+:::info
+
+Complex sharding with any success mode doesn't make sense when user asks for N number of tests to run explicitly, and we pass on the first one.
+
+:::
+
+#### All success
+Test passes if and only if all the executions are passing. This mode works only if there are no retries, i.e. no complex flakiness strategy, no retry strategy.
+
+<Tabs>
+<TabItem value="YAML" label="Marathonfile">
+
+```yaml
+executionStrategy:
+  mode: ALL_SUCCESS
+```
+
+</TabItem>
+<TabItem value="kts" label="Kotlin DSL">
+
+```kotlin
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ALL_SUCCESS)
+}
+```
+
+</TabItem>
+<TabItem value="groovy" label="Groovy DSL">
+
+```groovy
+marathon {
+  executionStrategy = ExecutionStrategyConfiguration(ExecutionMode.ALL_SUCCESS)
+}
+```
+
+</TabItem>
+</Tabs>
+
+:::info
+
+Adding retries with retry/flakiness strategies means users wants to trade-off cost for reliability, i.e. add more retries and pass if one 
+of test retries passes, so retries only make sense for any success mode. 
+
+When we use all success mode it means user wants to verify each test with a number of tries (they are not retries per se) and pass only if 
+all of them succeed. This is the case when fixing a flaky test or adding a new test, and we want to have a signal that the test is fixed/not flaky.
+
+:::
+
+#### Fast execution mode
+When test reaches a state when a decision about it's state can be made marathon can remove additional in-progress retries.
+This decision point is different depending on the execution mode used. Let's walk through two examples.
+
+Assume any success strategy is used and 100 retries are scheduled for a test A via flakiness strategy. Let's say first 3 failed and the 4th attempt succeeded. At 
+this point the test should already be considered passed since any success out of all retries leads to the result by definition of any successs
```

Comment:
Except for the first use of "any success" on this line, is it worth changing all instances of "any success" & "all success" to `ANY_SUCCESS` & `ALL_SUCCESS` for easier reading?